### ISSUE TITLE
Customize the hyphenation bounds?

### ISSUE THREAD

**Issue by baskerville** [tapeinosyne/hyphenation#20]
The hyphenation bounds are currently [hardcoded](https://github.com/tapeinosyne/hyphenation/blob/54ff3481be8327ce715a6091cde53a0bdd66eb0a/hyphenation_commons/src/language.rs#L50). Would it be possible to load the bounds from a file instead? For example, a file named `en-us.bnd.txt` with content `2 3` would define the bounds for the `EnglishUS` language.

**Comment by tapeinosyne** [comment]
Although default values are indeed hard-coded, the hyphenation boundaries are a [public field](https://github.com/tapeinosyne/hyphenation/blob/54ff3481be8327ce715a6091cde53a0bdd66eb0a/hyphenation_commons/src/dictionary.rs#L36) of the dictionary type and can be changed freely.  I will add a note in the documentation to this effect.

**Comment by baskerville** [comment]
I can get close to the expected output (`[6, 7]`), with:
```Rust
en_us.opportunities_within(word, en_us.boundaries(word).unwrap())
```
which returns `[6]`.

**Comment by baskerville** [comment]
In fact it seems to occur only when a word has a known hyphenation:
```Rust
    fn opportunities(&'h self, lowercase_word : &str) -> Vec<Self::Opportunity> {
        match self.boundaries(lowercase_word) {
            None => vec![],
            Some(mins) => {
                match self.exact(lowercase_word) {
                    None => self.opportunities_within(lowercase_word, mins),
                    Some(known) => known
                }
            }
        }
    }
```
You can't just return `known` here, you have to take `mins` into account.

**Comment by baskerville** [comment]
The reason `opportunities_within` returns `[6]` instead of `[6, 7]`, in the above example, is because it doesn't rely on `self.exact`.

**Comment by baskerville** [comment]
> the current behavior is counter-intuitive at best and plainly broken at worst

I concur.

**Comment by tapeinosyne** [comment]
Fixed in b5f526f90ead1e135bddce083e3d27d1e409194c.

Exceptions now respect hyphenation boundaries. The old behavior can be replicated by manually calling the internal method `exception_within` with (0, word.len()) as custom boundaries.
